submariner-operator/move-subctl-to-its-own-repo.md

```diff
@@ -0,0 +1,55 @@
+# Move subctl to its own repository
+
+## Summary
+
+Currently, `subctl` code lies in `submariner-operator` repository. Both subctl and operator require different set of dependencies which
+might conflict will improving subctl/operator. For eg, subctl can't be go install-ed currently because the operator dependencies don't allow
+it. It also makes operator upgrades difficult.
+
+## Proposal
+
+subctl, soon to be a library itself, should be moved out of submariner-operator repositoryto its own repo. Everything specific to the subctl
+is moved to the new subctl repo. subctl repositorydepends on the Operator repo. Operator repositoryshould contain things that the operator
+framework generates/needs.
+
+### Pros
+
+* Operator framework upgrades would be simpler.
+* It's logically correct to separate operator and a library as they are 2 different deliverables and provides different ways to install
+Submariner.
+* Testing criterias are different. We don't need subctl related GHAs for operator and vice versa.
+* Development approach is different. One is a CLI and another is a framework.
+* submariner-addon integration might be simplified.
+
+### Cons
+
+* A new repositoryto maintain.
+
```

Comment:
we are currently testing Operator with subctl. Now we will do the same but in new subctl repo.

Reply:
This is a valid concern, but it’s not much different from the other repositories which also use `subctl` during their tests. The operator is perhaps more likely than the other projects to require concomitant changes to `subctl`, but we’d need to deal with that situation anyway to ensure that upgrades are always possible.

@Jaanki I think it would be useful to mention this:

```suggestion
Splitting up the operator and `subctl` will require care when implementing changes in either which affect the other.
```

Reply:
> This is a valid concern, but it’s not much different from the other repositories which also use subctl during their tests. The operator is perhaps more likely than the other projects to require concomitant changes to subctl, but we’d need to deal with that situation anyway to ensure that upgrades are always possible.

Yes but the inter-connected-ness here is much more profound than in other repos, and the cyclical nature of things could get very confusing (esp. if a change requires a patch in both repos simultaneously)

Reply:
Which is pretty much what I said.

This makes explicit something which is already true. Currently, `subctl` and the operator are developed in the same repository, but they are consumable separately: one can (try to) use any version of `subctl` with any version of the operator. It’s simpler to develop against a single, common, repository, but splitting the repositories shouldn’t introduce any insurmountable challenges, and it will force us to actually think about changes that involve both projects.